babel/numbers.py

```diff
@@ -20,46 +20,104 @@
 #  - http://www.unicode.org/reports/tr35/ (Appendix G.6)
 import re
 from datetime import date as date_, datetime as datetime_
+from itertools import chain
 
-from babel.core import default_locale, Locale, get_global
+from babel.core import default_locale, Locale, get_global, UnknownLocaleError
 from babel._compat import decimal
+from babel.localedata import locale_identifiers
 
 
 LC_NUMERIC = default_locale('LC_NUMERIC')
 
 
-def get_currency_name(currency, count=None, locale=LC_NUMERIC):
+def list_currency(locale=None):
+    """ Return a list of normalized currency codes.
+
+    .. versionadded:: 2.5.0
+
+    :param locale: filters returned currency codes by the provided locale.
+                   Expected to be a locale instance or code. If no locale is
+                   provided, returns the list of all currencies from all
+                   locales.
+    """
+    # Get locale-scopped currencies.
+    if locale:
+        return Locale.parse(locale).currencies.keys()
+    # Aggregate currency codes from all locales.
+    return set(chain.from_iterable([
+        Locale.parse(l).currencies.keys() for l in locale_identifiers()]))
```

Comment:
I'm not too happy with that brute force approach either. At the time I rote that wasn't really certain about the completeness of the CLDR data, hence the way I took.

I think I'll first ensure with a unit-test that, as you said, that there's no specific language providing a currency code any other languages don't know about. If we can verify this assumption then I'll try to dig into the CLDR import code and see what I can do.